modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/itemselector/plugin.js

```diff
@@ -56,23 +56,23 @@
 				return;
 			}
 
-			const url = imageSrc.url ? imageSrc.url : imageSrc;
-
 			const editorContentDocument =
 				!editor.window.$.AlloyEditor &&
 				!editorContent.id.endsWith('BalloonEditor')
 					? editorContent.querySelector('iframe').contentDocument
 					: editorContent;
 
 			const imgElement = editorContentDocument.querySelector(
-				`img[src='${url}']`
+				`div[data-fileentryid="${imageSrc.fileEntryId}"]`
```

Comment:
I don't see why this selector needs to change. Most likely I'm missing some piece of context, let's discuss that tomorrow @kresimir-coko

Reply:
@dsanz IIRC that locator didn't work, it's because `url` contains a unique identifier of the portlet, which I have no way of knowing at this point in the code (AFAIK).

I prefer the new locator as it's directly tied to the `imageSrc` (`fileentryid` === `fileEntryId`)

That's actually one of the points where the bug was hidden by the `adaptivemedia` plugin hijacking the callback, that selector was never reached.

Reply:
@dsanz it works with the old img selector, I've reverted the change in the most recent commit